### PR TITLE
refactor mongodb metadata.yaml file

### PR DESCRIPTION
**Pull request by mukundansundar**
# Description

Refactor mongodb metadata file, to move `connectionString` into authentication profile and also `username`, `password`, `host`, `server` and `params` into another authentication profile. 

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #2658 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: state/mongodb/metadata.yaml [hunk 1]
```diff
@@ -17,34 +17,49 @@ capabilities:
...
-    # Required if server is not set
-    required: false
-    description: |
-      The host to connect to. One of `server` and `host` is required.
-    example: '"mongo-mongodb.default.svc.cluster.local:27017"'
-  - name: connectionstring
-    # Required if host and server are not set
-    required: false
+authenticationProfiles:
+  - title: "Connection string based authentication"
```
Comment (ItalyPaleAle) [inline state/mongodb/metadata.yaml:21]:
```suggestion
  - title: "Connection string"
```

File: state/mongodb/metadata.yaml [hunk 2]
```diff
@@ -17,34 +17,49 @@ capabilities:
...
-    example: '"mongodb://localhost:27017"'
-  - name: username
+      Authenticate using a connetion string.
+    metadata:
+      - name: connectionString
+        required: true
+        description: |
+          The connection string to use. One of `server` or `host` or `connectionstring` is required.
+        example: '"mongodb://localhost:27017"'
+  - title: "Username and password based authentication"
```
Comment (ItalyPaleAle) [inline state/mongodb/metadata.yaml:30]:
```suggestion
  - title: "Username and password"
```

File: state/mongodb/metadata.yaml [hunk 3]
```diff
@@ -17,34 +17,49 @@ capabilities:
...
     description: |
-      The connection string to use. One of `server` or `host` or `connectionstring` is required.
-    example: '"mongodb://localhost:27017"'
-  - name: username
+      Authenticate using a connetion string.
+    metadata:
+      - name: connectionString
+        required: true
+        description: |
+          The connection string to use. One of `server` or `host` or `connectionstring` is required.
```
Comment (ItalyPaleAle) [inline state/mongodb/metadata.yaml:28]:
```suggestion
          The connection string to use. One of `server`, `host`, and `connectionstring` is required.
```

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Thanks! Just a couple of small tweaks

File: state/mongodb/metadata.yaml (hunk 3 above)
Comment (mukundansundar) [reply state/mongodb/metadata.yaml:28]: changed to 
`The connection string to use. One of `server`, `host`, or `connectionstring` is required.`

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**